src/components/login/Login.tsx

```diff
@@ -1,15 +1,36 @@
+// import { useState } from 'react'
+
+import { FormEvent, useState } from 'react'
+
+import { useNavigate } from 'react-router-dom'
+
+import LoginCredentials from '../../interfaces/LoginCredentials'
+import { getLogin } from '../../services/auth.service'
+
 import './Login.css'
 
-function Login() {
+function Login(props: { setIsLogged : React.Dispatch<React.SetStateAction<boolean>> }) {
+  const [ user, setUser ] = useState({ email: '', password: '' })
+  const navigate = useNavigate()
+
+  const handleSubmit = (e : FormEvent) => {
+    e.preventDefault()
+    const request: LoginCredentials = {
+      email: user.email,
+      password: user.password
+    }
+    getLogin(request).then(response => {
+      localStorage.setItem('token', response.data.accessToken)
+      props.setIsLogged(true)
+      navigate('/')
+    }).catch(error => {alert(error)})
+  }
+
   return (
-    <form className="form" action="" method="">
-      <label>Username</label>
-      <input className = "login-input" type="text" placeholder="Enter Username" name="uname" required/>
-      <br/>
-      <label>Password</label>
-      <input className = "login-input" type="password" placeholder="Enter Password" name="psw" required/>
-      <br/>
-      <button type="submit">Login</button>
+    <form className="form" onSubmit={handleSubmit}>
+      <input className = "login-input" type="email" placeholder="Email" name="uname" onChange={(e) => setUser(prevState => ({ ...prevState, email: e.target.value }))} required/>
+      <input className = "login-input" type="password" placeholder="Password" name="psw" onChange={(e) => setUser(prevState => ({ ...prevState, password: e.target.value }))} required/>
+      <input className = "submit" type="submit" value="Sign In" />
     </form>
```

Comment:
Maybe break this in multiple lines. I like to stay between 100 and 120 characters per line.

Reply:
Btw I have trouble reading it formatted in this way. Also, if you have more than 3 attributes per element, the good thing to do is to break them in multiple lines. (maybe even less, but that's up to debate)

Reply:
Also, you can move implementation of handler functions for onChange events from tsx code, it is clener and easier to maintain